internal/manifests/config/configmap.go

```diff
@@ -160,9 +160,15 @@ func BuildConfigs(tempo v1alpha1.Microservices, params Params) (*corev1.ConfigMa
 		s3Endpoint = strings.TrimPrefix(s3Endpoint, "https://")
 	}
 
+	globalRetention := tempo.Spec.Retention.Global.Traces
+	if globalRetention == "" {
```

Comment:
This should be done in the defaulting webhook and the type of `tempo.Spec.Retention.Global.Traces` should be probably golang duration and not string to enable validation.

Reply:
fair point :+1:

Reply:
changed the trace type from `string` to `time.Duration`. Its a good thing, since they use `time.Duration` also in the tempo config.